lib/Net/SIP/Simple/Call.pm

```diff
@@ -452,9 +452,17 @@ sub dtmf {
 	    $type = $self->{param}{sdp_peer}
 		&& $self->{param}{sdp_peer}->name2int('telephone-event/8000','audio');
 	} elsif ( $m eq 'audio' ) {
-	    $type = $self->{param}{sdp_peer}
-		&& $self->{param}{sdp_peer}->name2int('PCMU/8000','audio')
-		|| 0; # default id for PCMU/8000
+	    my $sdp_peer = $self->{param}{sdp_peer};
+	    if ($sdp_peer) {
+		$type = $sdp_peer->name2int('PCMU/8000','audio') unless defined $type;
+		$type = $sdp_peer->name2int('PCMA/8000','audio') unless defined $type;
+		if (!defined $type) {
+			my $media = $sdp_peer->get_media();
+			my ($fmt) = map { ($_->{media} eq 'audio') ? $_->{fmt} : () } @$media;
+			($type) = grep { $_ == 0 || $_ == 8 } @$fmt; # Only PCMU=0 and PCMA=8 are supported
+		}
+	    }
+	    $type = 0 unless defined $type; # default id for PCMU/8000
```

Comment:
This could again by done simpler with `$type //= 0`.

Reply:
Done!